frontends/election-manager/src/components/format_usb_modal.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { assert, throwIllegalValue } from '@votingworks/utils';
+
+import {
+  Button,
+  Modal,
+  Prose,
+  isElectionManagerAuth,
+  isSystemAdministratorAuth,
+  LinkButton,
+  Loading,
+  UsbImage,
+} from '@votingworks/ui';
+import { AppContext } from '../contexts/app_context';
+
+export interface FormatUsbModalProps {
+  onClose: () => void;
+}
+
+type ModalState = 'init' | 'confirm' | 'formatting' | 'done' | 'error';
+
+export function FormatUsbModal({ onClose }: FormatUsbModalProps): JSX.Element {
+  const { usbDrive, auth } = useContext(AppContext);
+  assert(isSystemAdministratorAuth(auth) || isElectionManagerAuth(auth));
+  const userRole = auth.user.role;
+
+  const [state, setState] = useState<ModalState>('init');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const format = useCallback(async () => {
+    setState('formatting');
+    try {
+      await usbDrive.format(userRole, { action: 'eject' });
+      setState('done');
+    } catch (error) {
+      setState('error');
+      setErrorMessage((error as Error).message);
+    }
+  }, [usbDrive, userRole]);
+
+  // Handle USB drive being removed during stages of the  process
+  useEffect(() => {
+    if (usbDrive.status === 'absent') {
+      setState('init');
+    }
+  }, [state, usbDrive.status]);
+
+  if (state === 'error') {
+    return (
+      <Modal
+        content={
+          <Prose>
+            <h1>Failed to Format USB Drive</h1>
+            <p>Failed to format USB drive: {errorMessage}</p>
+          </Prose>
+        }
+        onOverlayClick={onClose}
+        actions={<Button onPress={onClose}>Close</Button>}
+      />
+    );
+  }
+
+  if (state === 'done') {
+    return (
+      <Modal
+        content={
+          <Prose>
+            <h1>USB Drive Formatted</h1>
+            <p>
+              USB drive successfully re-formatted. It is now ready to use with
```

Comment:
I think "reformat" is general spelled without the hyphen. If you agree we should remove the hyphen, there are a few other places that would also need to be updated

Reply:
Agree, removed in four places.